File: day7/day7es2.py
class Studente:
    def __init__(self, nome, matricola):
        self.nome = nome
        self.matricola = matricola
        self.voti = []

    def aggiungiVoto(self, voto):
        self.voti.append(voto)

    def calcolaMedia(self):
        if not self.voti:
            return 0
        return sum(self.voti) / len(self.voti)


studente2 = Studente("Giulia Bianchi", 54321)
studente2.aggiungiVoto(28)
studente2.aggiungiVoto(30)
print(f"La lista voti di Giulia è: {studente2.voti}")

File: day7/day7es2.js
class Studente {
    constructor(nome, matricola) {
        this.nome = nome;
        this.matricola = matricola;
        this.voti = [];
    }

    aggiungiVoto(voto) {
        this.voti.push(voto);
    }

    calcolaMedia() {
        if (this.voti.length === 0) {
            return 0;
        }
        const somma = this.voti.reduce((acc, voto) => acc + voto, 0);
        return somma / this.voti.length;
    }
}


const studente2 = new Studente("Giulia Bianchi", 54321);
studente2.aggiungiVoto(28);
studente2.aggiungiVoto(30);
console.log(`La lista voti di Giulia è: ${studente2.voti}`);